pkg/events/events.go

```diff
@@ -0,0 +1,376 @@
+		name     string
+		informer cache.SharedIndexInformer
+	}{
+		{"Events", w.eventInf},
+	}
+
+	for _, inf := range informers {
+		if !cache.WaitForCacheSync(stopc, inf.informer.HasSynced) {
+			log.Errorf("failed to sync %s cache", inf.name)
+			ok = false
+		} else {
+			log.Debugf("successfully synced %s cache", inf.name)
+		}
+	}
+	if !ok {
+		return errors.New("failed to sync caches")
+	}
+	log.Info("successfully synced all caches")
+	return nil
+}
+
+func worker(queue workqueue.RateLimitingInterface, syncFn func(string) error) {
+	for processNext(queue, syncFn) {
+	}
+}
+
+func processNext(queue workqueue.RateLimitingInterface, syncFn func(string) error) bool {
+	key, quit := queue.Get()
+	if quit {
+		return false
+	}
+
+	defer queue.Done(key)
+
+	err := syncFn(key.(string))
+	if err == nil {
+		queue.Forget(key)
+		return true
+	}
+
+	log.Error(err, "sync failed")
+	queue.AddRateLimited(key)
+
+	return true
+}
+
+func (w *Watcher) addHandlers() {
+	w.eventInf.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: w.handleEventAdd,
+		UpdateFunc: func(old, cur interface{}) {
+			w.handleEventAdd(cur)
+		},
+	})
+}
+
+func (w *Watcher) handleEventAdd(obj interface{}) {
+	key, ok := keyFunc(obj)
+	if !ok {
+		return
+	}
+
+	log.Debugf("Event added: %s", key)
+
+	enqueue(w.eventQueue, key)
+}
+
+func (w *Watcher) syncEvent(key string) error {
+	obj, exists, err := w.eventInf.GetIndexer().GetByKey(key)
+	if err != nil {
+		return err
+	}
+
+	if !exists {
+		return nil
+	}
+
+	e := obj.(*v1.Event)
+	log.Debugf("%s/%s: %d", e.Namespace, e.Name, e.Count)
+
+	w.lock.Lock()
+	defer w.lock.Unlock()
+
+	staleEvent := false
+
+	oldev, existingEvent := w.eventCache[e.Name]
+	if existingEvent {
+		log.Debugf("Found existing event: %s", e.Name)
+		if e.Count <= oldev.ev.Count {
+			log.Debugf("Existing event has not changed: %s", e.Name)
+			staleEvent = true
+		}
+	}
+
+	if !existingEvent || !staleEvent {
+		log.Debugf("Adding new event %s/%s: %d", e.Namespace, e.Name, e.Count)
+		w.eventCache[e.Name] = &evstore{
+			createdAt: time.Now(),
+			ev:        e,
+		}
+	}
+
+	/*log.Debugf("\nns: %s, \nname: %s, \nreason: %s, \nkind: %s, \ninv obj: %s, \ninv obj ns: %s,\nmsg: %s, \ncnt: %d, \nevtm: %s, \ncn: %s, \nrc: %s, \nri: %s, \nac: %s",
+	e.Namespace,
+	e.Name,
+	e.Reason,
+	e.InvolvedObject.Kind,
+	e.InvolvedObject.Name,
+	e.InvolvedObject.Namespace,
+	e.Message,
+	e.Count,
+	e.EventTime,
+	e.ClusterName,
+	e.ReportingController,
+	e.ReportingInstance,
+	e.Action)*/
+
+	return nil
+}
+
+func (w *Watcher) deleteEvent(retain int) {
+	w.lock.Lock()
+	defer w.lock.Unlock()
+	t := time.Now()
+	for n, ev := range w.eventCache {
+		if t.After(ev.createdAt.Add(time.Duration(retain) * time.Minute)) {
+			log.Debugf("Deleting old entry: %s/%s: %d", ev.ev.Namespace, ev.ev.Name, ev.ev.Count)
+			delete(w.eventCache, n)
+		}
+	}
+}
+
+func (w *Watcher) deleteEvents(stop, done chan bool) {
+	loop, err := strconv.Atoi(os.Getenv("EVENT_CHECK_SECS"))
```

Comment:
Consider using a default value as second parameter.

Reply:
os.Getenv does not have a second default parameter, its there for python not for golang